transport/http.go

```diff
@@ -71,6 +71,37 @@ var (
 	defaultServerTimeout = 30 * time.Second
 )
 
+// HTTPTransportOptions for the HTTPTransport.
+type HTTPTransportOptions struct {
+	// APIKey holds the APIKey base64-encoded string.
+	APIKey string
+
+	// SecretToken holds the secret token configured in the APM Server.
+	SecretToken string
+
+	// ServerTLSCert can be set if you have configured your APM Server with a
+	// self signed TLS certificate, or you want to pin the server certificate,
+	// specify the path to the PEM-encoded certificate with `ServerCACert`.
```

Comment:
I'm having a hard time parsing this sentence. What's ServerCACert got to do with this? If the server certificate is being pinned, then there's no point in setting the CA cert as well is there?

Also, naming quibble: for consistency either this one should be ServerCert or the other one should be ServerTLSCACert. They're both to do with TLS as much as the other. I'd probably go with ServerCert, seeing as the env var is `ELASTIC_APM_SERVER_CERT`.